### PR TITLE
Find resolvers locally when Bower is installed globally

### LINKED ISSUE #1919: Custom Resolvers need to be globally installed.
I was trying to make my own resolver. 

I found that only when I installed my **Custom Resolver globally using `npm -g`** did it work.

I'm on Windows. On trying to use local resolvers, I am getting:

`bower MODULE_NOT_FOUND      Cannot find module 'test-resolver'`

### PR DESCRIPTION
**Pull request by wrumsby**
PoC implementation for review and feedback.

Fixes #1919

### CODE REVIEW

**Comment by sheerun** [comment]
we need to use config.cwd instead `./`

File: lib/core/resolverFactory.js [hunk 1]
```diff
@@ -56,8 +56,21 @@ function getConstructor(decEndpoint, options, registryClient) {
...
         var resolverPromises = resolverNames.map(function (resolverName) {
-            var resolver = resolvers[resolverName]
-                || pluginResolverFactory(require(resolverName), options);
+            var resolver = resolvers[resolverName];
+
+            if (resolver === undefined) {
+                try {
+                    resolver = pluginResolverFactory(require(resolverName), options);
+                } catch (e) {
+                    if (e.code === 'MODULE_NOT_FOUND') {
```
Comment (sheerun) [inline lib/core/resolverFactory.js]: re-throw otherwise?

Comment (sheerun) [inline lib/core/resolverFactory.js]: also maybe we should check for the name of module that failed to load here

**Comment by wrumsby** [comment]
Sorry, I got distracted - will try to get back to this over the next few days.

**Comment by sheerun** [comment]
Fixed with https://github.com/bower/bower/commit/f4cb047e9df55f03e9971ce4f4341c41d99339e3 and will be released in bower 1.7.8

**Pull request closed without merging**